src/integrity-checker.js

```diff
@@ -253,7 +259,7 @@ export default class InstallationIntegrityChecker {
       lockfile,
       patterns,
       Object.assign({}, flags, {checkFiles: false}), // don't generate files when checking, we check the files below
-      loc.locationFolder,
+      await this._getModuleFolder(),
```

Comment:
https://github.com/yarnpkg/yarn/pull/3478/files#diff-9aeebf9c5a183aebcbff42975e733805R64

It's the old behavior to find the module folder, extracted into its own function. It's required, since until now the `files` entry was populated with the file entries starting from the `.yarn-integrity` location. Now that they are separated, the logic needs to be separated as well.